src/main/java/com/tw/go/plugin/OAuthLoginPlugin.java

```diff
@@ -207,12 +207,17 @@ private GoPluginApiResponse handleSetupLoginWebRequest(GoPluginApiRequest goPlug
             Properties oauthConsumerProperties = new Properties();
             oauthConsumerProperties.put(provider.getConsumerKeyPropertyName(), pluginSettings.getConsumerKey());
             oauthConsumerProperties.put(provider.getConsumerSecretPropertyName(), pluginSettings.getConsumerSecret());
+            String scopeprop = provider.getAuthScopePropertyName();
+            Permission perm = Permission.AUTHENTICATE_ONLY;
+            if ( scopeprop != null && scopeprop.length() > 0 ) {
+                perm = Permission.CUSTOM;
+                oauthConsumerProperties.put(provider.getAuthScopePropertyName(), provider.getAuthScopePropertyValue());
+            }
```

Comment:
Do you feel it's better to just do `Provider#getPermissions()` and `Provider#configure(Properties oauthConsumerProperties)` to allow flexibility for plugins in the future?

/cc @srinivasupadhya